sdk/messaging/azeventhubs/processor.go

```diff
@@ -0,0 +1,349 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package azeventhubs
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	azlog "github.com/Azure/azure-sdk-for-go/sdk/internal/log"
+)
+
+type ProcessorStrategy string
+
+const (
+	// ProcessorStrategyBalanced will attempt to claim a single partition at a time, until each active
+	// owner has an equal share of partitions.
+	// This is the default strategy.
+	ProcessorStrategyBalanced ProcessorStrategy = "balanced"
+
+	// ProcessorStrategyGreedy will attempt to claim as many partitions at a time as it can, ignoring
+	// balance.
+	ProcessorStrategyGreedy ProcessorStrategy = "greedy"
+)
+
+// NewProcessorOptions are the options for the NewProcessor
+// function.
+type NewProcessorOptions struct {
```

Comment:
Can you add descriptions for the default values for these fields?

Reply:
Added.